npx tsc fails due to body.Host

Hi there,

I recently forked the repository since I'm interested in adding some minor functionality.
Yet, I'm currently unable to build it. Running `npm run prepare` results in
```
> remarkable-typescript@1.0.5 prepare /media/fernando/shared/reMarkable-typescript
> npx tsc

src/remarkable.ts:93:39 - error TS2339: Property 'Host' does not exist on type 'string'.

93     this.storageUrl = `https://${body.Host}`;
                                         ~~~~

src/remarkable.ts:100:42 - error TS2339: Property 'Host' does not exist on type 'string'.

100     this.notificationUrl = `wss://${body.Host}`;
                                             ~~~~
```
It's a minor issue, but I was wondering if it's something that should be changed, since the line has been around since the initial commit, and the build on `npm` runs fine. How was that one compiled?

Cheers
Fernando

--- a/README.md
+++ b/README.md
@@ -24,27 +24,27 @@ const fs = require('fs');
 (async () => {
     /*
     * Create the reMarkable client
-    * Params: { token?: string }
+    * Params: { deviceToken?: string }
     * Returns: client: Remarkable
     */
     const client = new Remarkable();
 
     /*
-    * Register your reMarkable. You must do this first to pair your device if you didn't specify a token. This may take a few seconds to complete.
+    * Register your reMarkable and generate a device token. You must do this first to pair your device if you didn't specify a token. This may take a few seconds to complete. It seems that the deviceToken never expires.
     * Params: { code: string }
-    * Returns: token: string
+    * Returns: deviceToken: string
     */
-    const token = await client.register({ code: 'created code' });
+    const deviceToken = await client.register({ code: 'created code' });
 
-    // (optional) skip registration in the future with `new Remarkable({token})`
-    console.log(token);
+    // (optional) skip registration in the future with `new Remarkable({deviceToken})`
+    console.log(deviceToken);
 
     /*
-    * Refresh the token. For now, the token does not expire but this may change in the future
+    * (Re)generate a token from the deviceToken. This token, used to interact with storage, is different from the deviceToken. This function is automatically called when in the constructor and in register(). This token expires.
     * Params: none
     * Returns: token: string
     */
-    const newToken = await client.refreshToken();
+    await client.refreshToken();
 
     /*
     * List all items, files and folders.
@@ -81,6 +81,13 @@ const fs = require('fs');
     * Returns: Buffer
     */
     const zipFile = await client.downloadZip(pdfUploadedId);
+
+    /*
+    * Upload a ZIP file to your reMarkable (must be a supported reMarkable format). You can generate the ID using uuidv4.
+    * Params: name: string, id: string, zipFile: Buffer
+    * Returns: Buffer
+    */
+    const zipFileId = await client.uploadZip('My document name', ID: 'f831481c-7d2d-4776-922d-36e708d9d680', zipFile);
 })();
 ```
 

--- a/src/remarkable.ts
+++ b/src/remarkable.ts
@@ -46,18 +46,22 @@ const defaultPDFmetadata = {
 };
 
 type Props = {
-  token?: string;
+  deviceToken?: string;
 };
 
 export default class Remarkable {
   private token?: string;
+  private deviceToken?: string;
   private client: Got = got.extend(gotConfiguration);
   private storageUrl?: string;
   private notificationUrl?: string;
   private zip: JSZip;
 
-  constructor({ token }: Props = {}) {
-    if (token) this.setToken(token);
+  constructor({ deviceToken }: Props = {}) {
+    if (deviceToken) {
+      this.deviceToken = deviceToken;
+      this.refreshToken();
+    }
     this.zip = new JSZip();
   }
 
@@ -76,10 +80,10 @@ export default class Remarkable {
   }
 
   public async refreshToken() {
-    if (!this.token) throw new Error('You must register your reMarkable first');
+    if (!this.deviceToken) throw new Error('You must register your reMarkable first');
     const { body }: { body: string } = await got.post('https://my.remarkable.com/token/json/2/user/new', {
       headers: {
-        Authorization: `Bearer ${this.token}`,
+        Authorization: `Bearer ${this.deviceToken}`,
         'User-Agent': `remarkable-typescript/${pkgVersion}`,
       },
     });
@@ -89,14 +93,14 @@ export default class Remarkable {
 
   public async getStorageUrl({ environment = 'production', group = 'auth0|5a68dc51cb30df3877a1d7c4', apiVer = 2 } = {}) {
     if (this.storageUrl) return this.storageUrl;
-    const { body } = await this.client.get(`https://service-manager-production-dot-remarkable-production.appspot.com/service/json/1/document-storage?environment=${environment}&group=${group}&apiVer=${apiVer}`);
+    const body: { Host: string, Status: string} = await this.client.get(`https://service-manager-production-dot-remarkable-production.appspot.com/service/json/1/document-storage?environment=${environment}&group=${group}&apiVer=${apiVer}`).json();
     this.storageUrl = `https://${body.Host}`;
     return this.storageUrl;
   }
 
   public async getNotificationsUrl({ environment = 'production', group = 'auth0|5a68dc51cb30df3877a1d7c4', apiVer = 1 } = {}) {
     if (this.notificationUrl) return this.notificationUrl;
-    const { body } = await this.client.get(`https://service-manager-production-dot-remarkable-production.appspot.com/service/json/1/notifications?environment=${environment}&group=${group}&apiVer=${apiVer}`);
+    const body: { Host: string, Status: string} = await this.client.get(`https://service-manager-production-dot-remarkable-production.appspot.com/service/json/1/notifications?environment=${environment}&group=${group}&apiVer=${apiVer}`).json();
     this.notificationUrl = `wss://${body.Host}`;
     return this.notificationUrl;
   }
@@ -120,9 +124,10 @@ export default class Remarkable {
     return got.post('https://my.remarkable.com/token/json/2/device/new', {
       json: { code, deviceDesc, deviceId: deviceId || generateDeviceId() },
     })
-      .then(({ body }) => {
-        this.setToken(body);
-        return this.refreshToken();
+      .then(async ({ body }) => {
+        this.deviceToken = body;
+        await this.refreshToken();
+        return body;
       });
   }
 

